File: 392.Is_Subsequence/solution.py
def isSubsequence(s: str, t: str) -> bool:
    check_string = ""
    for letter in s:
        for l in t:
            if l == letter:
                check_string += l
                t = t.replace(l, "", 1)
                break
    return check_string == s


# case 1
print(isSubsequence("abc", "ahbgdc"))

# case 2
print(isSubsequence("axc", "ahbgdc"))

File: 392.Is_Subsequence/solution.js
// * Accepted

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  let tIndex = 0;
  let sIndex = 0;
  let str = "";
  while (tIndex < t.length) {
    let tPointer = t[tIndex];
    let sPointer = s[sIndex];
    if (tPointer === sPointer) {
      tIndex++;
      sIndex++;
      str += tPointer;
    } else {
      tIndex++;
    }
  }
  return str === s;
};

// case 1
console.log(isSubsequence("abc", "ahbgdc")); // ? true

// case 2
console.log(isSubsequence("axc", "ahbgdc")); // ? false
